src/transformers/modeling_utils.py

```diff
@@ -1117,31 +1137,17 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
         config.name_or_path = pretrained_model_name_or_path
 
         # Instantiate model.
-
         if is_deepspeed_zero3_enabled():
             import deepspeed
 
             logger.info("Detected DeepSpeed ZeRO-3: activating zero.init() for this model")
             # this immediately partitions the model across all gpus, to avoid the overhead in time
             # and memory copying it on CPU or each GPU first
-            with deepspeed.zero.Init(config=deepspeed_config()):
+            with deepspeed.zero.Init(config=deepspeed_config()) and no_init_weights(_disable=_no_fast_init):
                 model = cls(config, *model_args, **model_kwargs)
```

Comment:
```suggestion
            with deepspeed.zero.Init(config=deepspeed_config()):
                with no_init_weights(_disable=_no_fast_init):
                    model = cls(config, *model_args, **model_kwargs)
```

I'm not quite sure what happens when you `and` 2 context managers, I tried to find the authoritative answer, but I only found the spec that says it should be comma-separated. https://www.python.org/dev/peps/pep-0343/#specification-the-with-statement

But I propose to stack them vertically, so it's easy to see that `with deepspeed` is a stack around an identical entry from the `else` branch.